### ISSUE TITLE
Incorrect Indentation errors for expression switches with google_checks.xml

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#12817]
```bash
$ javac -fullversion
javac full version "20-ea+35-2342"
$ javac T.java

$ wget https://raw.githubusercontent.com/checkstyle/checkstyle/d02e867225c3eaa43dc681c2827ce89799dee88f/src/main/resources/google_checks.xml

$ cat T.java
class T {
  void doSomething(int i) {
    String s =
        switch (i) {
          case 1 -> "one";
          case 2 -> "two";
          default -> throw new AssertionError();
        };
  }
}
$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java -jar checkstyle-10.8.1-all.jar -c google_checks.xml T.java
Starting audit...
[WARN] /tmp/tmp.zK57404080/T.java:4:9: 'switch' has incorrect indentation level 8, expected level should be 4. [Indentation]
[WARN] /tmp/tmp.zK57404080/T.java:5:11: 'case' child has incorrect indentation level 10, expected level should be 6. [Indentation]
[WARN] /tmp/tmp.zK57404080/T.java:6:11: 'case' child has incorrect indentation level 10, expected level should be 6. [Indentation]
[WARN] /tmp/tmp.zK57404080/T.java:7:11: 'case' child has incorrect indentation level 10, expected level should be 6. [Indentation]
[WARN] /tmp/tmp.zK57404080/T.java:8:9: 'switch rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
```

**Describe what you expect in detail.**

I expected no errors to be produced, because the input is formatted in a way that complies with the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)

**Comment by Alexanderwar** [comment]
maybe you are having problem with the compiler i have executed the file and it got implemented corrected.May be you are using the early release version.The file runs correctly with Java(TM) SE Runtime Environment (build 19.0.2+7-44)
Java HotSpot(TM) 64-Bit Server VM (build 19.0.2+7-44, mixed mode, sharing).

**Comment by Alexanderwar** [comment]
`class T {
    void doSomething(int i) {
        String s =
            switch (i) {
                case 1 -> "one";
                case 2 -> "two";
                default -> throw new AssertionError();
            };
    }
    
    public static void main(String[] args) {
        T t = new T();
        t.doSomething(1); // calling the doSomething method with the argument 1
    }
}
`

**Comment by destebanm** [comment]
Hi!
I have the same issue with switch. 

I reported in https://github.com/google/google-java-format/issues/915

Thanks!

**Comment by deepanshu150895** [comment]
I am also facing this issue, 
error - 'switch' has incorrect indentation level 8, expected level should be 4. [Indentation]
`static void testMethod(int month) {
    int result =
        switch (month) {
          case JANUARY, JUNE, JULY -> 3;
          case FEBRUARY, SEPTEMBER, OCTOBER, NOVEMBER, DECEMBER -> 1;
          case MARCH, MAY, APRIL, AUGUST -> {
            yield month * 4;
          }
          default -> 0;
        };
  }`

**Comment by romani** [comment]
Fix is merged

**Comment by pnagy-cldr** [comment]
@romani I have bumped into this issue and using the currently latest 10.23.0 version did not solve it. In my case I use pattern matching in a lambda expression. It seems to me that there is no testcase for this. Here is an example code without my exact types:

```
    Comparator<BaseType> comparator = comparing(baseType ->
            switch (baseType) {
              case FirstSubType firstSubType -> firstSubType.getMethod();
              case SecondSubType secondSubType -> secondSubType.getMethod();
              default -> null;
            },
        Comparator.nullsLast(naturalOrder()));
```

Could you check this kind of scenario?

**Comment by romani** [comment]
@pnagy-cldr , please report new issue
